quic/s2n-quic-platform/src/message/cmsg.rs

```diff
@@ -4,173 +4,248 @@
 #![allow(clippy::unnecessary_cast)] // some platforms encode lengths as `u32` so we cast everything to be safe
 
 use core::mem::{align_of, size_of};
+use libc::cmsghdr;
 use s2n_quic_core::inet::{AncillaryData, ExplicitCongestionNotification};
 
+const fn size_of_cmsg<T: Copy + Sized>() -> usize {
+    unsafe { libc::CMSG_SPACE(size_of::<T>() as _) as _ }
+}
+
+const fn const_max(a: usize, b: usize) -> usize {
```

Comment:
I was going to suggest linking to https://github.com/rust-lang/rust/issues/92391 in a TODO, but seems like that is going in a [different direction](https://github.com/rust-lang/rust/issues/92391#issuecomment-1515911121) anyway